scripts/strip.py

```diff
@@ -0,0 +1,292 @@
+
+
+class SectionTable:
+    data: BytesIO
+    offset: int
+    size: int
+    entrySize: int
+    elfClass: EiClass
+    endianness: EiData
+
+    def __init__(self, b: BytesIO, e: ElfEhdr, ei: ElfEident):
+        self.data = b
+        self.offset = e.e_shoff
+        self.size = e.e_shnum * e.e_shentsize
+        self.entrySize = e.e_shentsize
+        self.elfClass = ei.get_class()
+        self.endianness = ei.get_endianness()
+
+    def __iter__(self):
+        self.entryOff = 0
+        self.data.seek(self.offset)
+        return self
+
+    def __next__(self):
+        if self.entryOff < self.size:
+            headers = {EiClass.ELFCLASS32: Elf32Shdr}
+            off = self.offset + self.entryOff
+            self.data.seek(off)
+            val = headers[self.elfClass].parse(self.data.read(self.entrySize), self.endianness)
+            self.entryOff += self.entrySize
+            return val, off
+        else:
+            raise StopIteration
+
+
+class RelocationTable:
+    data: BytesIO
+    offset: int
+    size: int
+    entrySize: int
+    type: ShType
+    elfClass: EiClass
+    endianness: EiData
+
+    def __init__(self, b: BytesIO, s: ElfShdr, ei: ElfEident):
+        self.type = ShType(s.sh_type)
+        if self.type not in [ShType.SHT_REL, ShType.SHT_RELA]:
+            raise ValueError(f"Section of type {self.type} is not relocation table!")
+        self.data = b
+        self.offset = s.sh_offset
+        self.size = s.sh_size
+        self.entrySize = s.sh_entsize
+        self.elfClass = ei.get_class()
+        self.endianness = ei.get_endianness()
+
+    def __iter__(self):
+        self.entryOff = 0
+        return self
+
+    def __next__(self):
+        if self.entryOff < self.size:
+            headers = {(EiClass.ELFCLASS32, ShType.SHT_REL): Elf32Rel}
+            off = self.offset + self.entryOff
+            self.data.seek(off)
+            val = headers[(self.elfClass, self.type)].parse(self.data.read(self.entrySize), self.endianness)
+            self.entryOff += self.entrySize
+            return val, off
+        else:
+            raise StopIteration
+
+
+class ElfParser:
+    data: BytesIO
+    ident: ElfEident
+    header: ElfEhdr
+    sections: SectionTable
+
+    def __init__(self, b: BytesIO):
+        self.data = b
+        self.ident = ElfEident.parse(self.data.read(16))
+        self.header = Elf32Ehdr.parse(self.data.read(36), self.ident.get_endianness())
+        self.sections = SectionTable(self.data, self.header, self.ident)
+
+    def get_sections(self) -> SectionTable:
+        return self.sections
+
+    def get_relocations(self, s: ElfShdr) -> RelocationTable:
+        return RelocationTable(self.data, s, self.ident)
+
+    def update(self, value: bytes, offset: int):
+        self.data.seek(offset)
+        self.data.write(value)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
```

Comment:
put it in function, all variables defined here are put in global scope (and You're redefining eg. `data` in at least a few other places (see pylint warnings))